src/Symfony/Component/Mailer/Bridge/Mailjet/Transport/MailjetApiTransport.php

```diff
@@ -177,4 +193,14 @@ private function getEndpoint(): ?string
     {
         return ($this->host ?: self::HOST).($this->port ? ':'.$this->port : '');
     }
+
+    private function castCustomHeader(string $value, string $type)
+    {
+        return match ($type) {
+            'bool' => filter_var($value, \FILTER_VALIDATE_BOOLEAN),
+            'int' => (int) $value,
+            'json' => json_decode($value, true, 2, \JSON_THROW_ON_ERROR),
+            default => $value,
```

Comment:
on second though, I'd prefer removing `default` and replace it by `'string'. I'd also prefer putting `'string'` explicitly in the const and removing the `?? 'string'` on L142L.